### DISCUSSION TITLE
Obliczenie prowizji od sprzedaży - wyróżnienie elastyczne

### DISCUSSION THREAD

**Discussion by krzysztofzylka** [allegro/allegro-api#4952] [Suggestion - verification]
Witam, próbuję obliczyć przybliżoną prowizję od sprzedaży za pomocą `/pricing/offer-fee-preview`, czy jest możliwość dodania do promocji  wyróżnienie elastyczne bo nie widzę takich informacji w dokumentacji https://developer.allegro.pl/documentation/#operation/calculateFeePreviewUsingPOST

edit: zależało by mi również na promocji promo_package

**Comment by AureliuszBrussy** [comment]
Zasób ten nie pozwala na oszacowanie opłat za wyróżnienia elastyczne czy pakiet promo. Jednocześnie dziękuję za sugestię - przeniosłem ją do zakładki Discussions, gdzie inni użytkownicy mogą także oddać swój głos na taką funkcjonalność.

**Comment by krzysztofzylka** [comment]
@AureliuszBrussy Porobiłem sobie testy przed Twoją odpowiedzią i udało mi się poprawnie wygenerować wycenę z wyróżnieniem elastycznym, czy to jest wykonane poprawnie?
`{
    "offer": {
        "category": {
            "id": "50768"
        },
        "parameters": [],
        "promotion": {
            "emphasized1d": true
        },
        "publication": {},
        "sellingMode": {
            "format": "BUY_NOW",
            "price": {
                "amount": 67.65,
                "currency": "PLN"
            }
        },
        "stock": {
            "avaliable": 695,
            "unit": "UNIT"
        }
    }
}`

w odpowiedzi dostaję:
`{
    "commissions": [
        {
            "name": "Prowizja od sprzedaży",
            "type": "commissionFee",
            "fee": {
                "amount": 6.09,
                "currency": "PLN"
            }
        },
        {
            "name": "Prowizja od sprzedaży oferty wyróżnionej",
            "type": "commissionFeatureFee",
            "fee": {
                "amount": 3.66,
                "currency": "PLN"
            }
        }
    ],
    "quotes": [
        {
            "name": "Opłata za wystawienie",
            "type": "listingFee",
            "fee": {
                "amount": 0.10,
                "currency": "PLN"
            },
            "cycleDuration": "PT240H",
            "classifiedsPackage": {
                "id": null
            }
        },
        {
            "name": "Opłata za wyróżnienie elastyczne",
            "type": "distinction1dFee",
            "fee": {
                "amount": 2.90,
                "currency": "PLN"
            },
            "cycleDuration": "PT24H",
            "classifiedsPackage": {
                "id": null
            }
        }
    ]
}`

**Comment by AureliuszBrussy** [comment]
Faktycznie, potwierdzam - dla `"emphasized1d": true` zwracamy informację o opłacie za wyróżnienie elastyczne. W przypadku `"promoPackage"` jest inaczej. Konsultuję to z zespołem odpowiedzialnym za tą usługę. Wrócę z informacją.

**Comment by AureliuszBrussy** [comment]
@krzysztofzylka na tą chwilę endpoint ten zapewnia obliczanie poprawnych wyceny tylko dla udokumentowanych, czyli następujących flag:

![image](https://user-images.githubusercontent.com/68948612/134530016-7770c5a3-42e2-4939-ae9d-5cdf55aa8865.png)

Typ **emphasized1d** i **promoOptions** nie powinny być obecnie używane.
